actors/builtin/market/market_actor.go

```diff
@@ -639,14 +665,22 @@ func deleteDealProposalAndState(dealId abi.DealID, states *DealMetaArray, propos
 func ValidateDealsForActivation(
 	st *State, store adt.Store, dealIDs []abi.DealID, minerAddr addr.Address, sectorExpiry, currEpoch abi.ChainEpoch,
 ) (big.Int, big.Int, uint64, error) {
-
 	proposals, err := AsDealProposalArray(store, st.Proposals)
 	if err != nil {
 		return big.Int{}, big.Int{}, 0, xerrors.Errorf("failed to load dealProposals: %w", err)
 	}
 
-	seenDealIDs := make(map[abi.DealID]struct{}, len(dealIDs))
+	return validateAndComputeDealWeight(proposals, dealIDs, minerAddr, sectorExpiry, currEpoch)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Checks
+////////////////////////////////////////////////////////////////////////////////
 
+func validateAndComputeDealWeight(proposals *DealArray, dealIDs []abi.DealID, minerAddr addr.Address,
+	sectorExpiry abi.ChainEpoch, sectorActivation abi.ChainEpoch) (big.Int, big.Int, uint64, error) {
```

Comment:
Consider subbing `minimumDealStart` for `sectorActivation` since this parameter can be either deal activation or current epoch depending on how it's called.

Reply:
I considered it but left as is. I initially misunderstood your proposal and thought you meant to type `maximum` [epoch at which the deal could start]. I eventually realised how to interpret it your way, but decided to just leave it meaning "proposed activation epoch". Then the question became whether to drop `sector` from the name, I left it there for consistency with `sectorExpiry`, where it makes good sense.